tests/cache/test-vary-cache.php

```diff
@@ -323,13 +323,26 @@ public function get_test_data__set_group_for_user_invalid() {
 	}
 
 	/**
-	 * @dataProvider get_test_data__set_group_for_user_valid
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
 	 */
-	public function test__set_group_for_user_valid( $group, $value, $expected_result ) {
-		$this->markTestSkipped('Skip for now until cookie is set in hook');
-		$actual_result  = Vary_Cache::set_group_for_user( $group, $value );
+	public function test__set_group_for_user__valid() {
+		$actual_result = Vary_Cache::set_group_for_user( 'dev-group', 'yep' );
 
-		$this->assertEquals( $expected_result, $actual_result );
+		$this->assertTrue( $actual_result, 'Return value was not true' );
+
+		$this->assertEquals( [ 'dev-group' => 'yep' ], Vary_Cache::get_groups(), 'Groups did not match expected value' );
+
+		$this->assertTrue( self::get_vary_cache_property( 'should_update_group_cookie' ), 'Did not update group cookie' );
+
+		// Verify cookie actions were taken
+		add_action( 'vip_vary_cache_did_send_headers', function( $sent_vary, $sent_cookie ) {
+			$this->assertTrue( $sent_vary, 'Vary was not sent' );
+			$this->assertTrue( $sent_cookie, 'Cookie was not sent' );
+		}, 10, 2 );
+
```

Comment:
```suggestion
$this->assertEquals( 1, did_action( 'vip_vary_cache_did_send_headers' ) );
```

Reply:
Needs to actually run after the `do_action` call (done in e1a1ace8)